### PR TITLE
feat(idp): adjust delete idp endpoint

### PR DESCRIPTION
**Pull request by Phil91**
## Description

* enable delete of managed idps
* enable deactivation of managed idps

## Why

to have the possibility to delete or deactivate the managed idps

## Issue

N/A - Jira Issue: CPLP-3329

## Corresponding CD PR

[#135](https://github.com/eclipse-tractusx/portal-cd/pull/135)

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Phil91** [comment]
before merging this PR, #379  should be merged

File: src/portalbackend/PortalBackend.DBAccess/Repositories/IdentityRepository.cs [hunk 1]
```diff
@@ -51,4 +51,15 @@ public IdentityRepository(PortalDbContext context)
                 x.IdentityTypeId,
                 x.CompanyId))
             .SingleOrDefaultAsync();
+
+    /// <inheritdoc />
+    public IAsyncEnumerable<(string Email, string? FirstName, string? LastName)> GetCompanyUserEmailForIdentityIdsWithoutOwnerAndRoleId(IEnumerable<Guid> userRoleIds, IEnumerable<Guid> identityIds) =>
+        _context.CompanyUsers
+            .Where(x =>
+                x.Identity!.Company!.IdentityProviders.Any(idp => identityIds.Contains(idp.Id) && idp.OwnerId != x.Identity!.CompanyId) &&
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/IdentityRepository.cs]: this is comparing apples to pears (identityIds to idpIds).

Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/IdentityRepository.cs]: solved

**Review by ntruchsess — CHANGES_REQUESTED** [review]
when setting users to DELETED the assigned Roles must be removed as well. As a consequence the role-based email-service cannot send emails to those users afterwards. That means the Names and email-addresses for the email-service must be determined before deleting the users

**Comment by Phil91** [comment]
> when setting users to DELETED the assigned Roles must be removed as well. As a consequence the role-based email-service cannot send emails to those users afterwards. That means the Names and email-addresses for the email-service must be determined before deleting the users

added the deletion of the assigned roles

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**